### PR TITLE
[DO NOT MERGE] Adding north-south api for PrP

### PR DESCRIPTION
**Pull request by reamkazaz**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by anuchandy — CHANGES_REQUESTED** [review]
@reamkazaz please use space for indentation, indentation for files has to be 4 spaces. 

You can open the  existing security.json file in upstream master in your editor and enable white space rendering (e.g. open json in VS Code then View -> Toggle Render Whitespace) to see this.

**Comment by anuchandy** [comment]
it seems we're still not using whitespace for tab.

tab

In swagger in upstream master uses whitespaces.

tab1

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json [hunk 1]
```diff
@@ -1437,6 +1437,437 @@
...
+                "x-ms-examples": {
+                    "List all north-south hardening resources for a subscription": {
+                        "$ref": "./examples/NorthSouthHardenings/ListAllNorthSouthHardenings_example.json"
+                    }
+                },
+                "description": "Gets a list of north-south hardening resources for the subscription.",
+                "tags": [
+                    "NorthSouthHardenings"
+                ],
+                "operationId": "NorthSouthHardenings_ListAll",
```
Comment (anuchandy) [inline specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: Naming convention for method that list resource in subscription level is `List` so operation id should be `NorthSouthHardenings_List`

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json [hunk 2]
```diff
@@ -1437,6 +1437,437 @@
...
+                "x-ms-examples": {
+                    "List north-south hardening resources for a resource group": {
+                        "$ref": "./examples/NorthSouthHardenings/ListNorthSouthHardenings_example.json"
+                    }
+                },
+                "description": "Gets a list of north-south hardening resources for the resource group.",
+                "tags": [
+                    "NorthSouthHardenings"
+                ],
+                "operationId": "NorthSouthHardenings_List",
```
Comment (anuchandy) [inline specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: Naming convention for method that list resource in resource-group level is `ListByResourceGroup` so operation id should be `NorthSouthHardenings_ ListByResourceGroup`

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json [hunk 3]
```diff
@@ -1437,6 +1437,437 @@
...
+                "x-ms-examples": {
+                    "Put a north-south hardening policy": {
+                        "$ref": "./examples/NorthSouthHardenings/PutNorthSouthHardeningsPolicies_example.json"
+                    }
+                },
+                "tags": [
+                    "NorthSouthHardenings"
+                ],
+                "description": "Put a north-south hardening policy",
+                "operationId": "NorthSouthHardeningsPolicies_Put",
```
Comment (anuchandy) [inline specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: If this operation is for creating or updating a policy resource then the operation id should be `NorthSouthHardeningsPolicies_CreateOrUpdate`

**Review by anuchandy — CHANGES_REQUESTED** [review]

**Comment by anuchandy** [comment]
@Ream Kazaz is there any more change coming? if not we can request ARM review since this PR introduces a set of new APIs.

**Comment by reamkazaz** [comment]
That would be all.
Please request Arm review for us.
Thanks

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json (hunk 1 above)
Comment (reamkazaz) [reply specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: Done

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json (hunk 2 above)
Comment (reamkazaz) [reply specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: Done

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json (hunk 3 above)
Comment (reamkazaz) [reply specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: Done

**Comment by reamkazaz** [comment]
@anuchandy After adding ARM reviewers do you mind adding the [Do not merge] label for us?

**Comment by anuchandy** [comment]
@reamkazaz - Requested ARM review. We will keep the same label. Once the service is deployed to production you can remove the "don't merge" text from PR label.

**Review by anuchandy — APPROVED** [review]

**Comment by anuchandy** [comment]
@ravbhatnagar can you take a look?

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json [hunk 4]
```diff
@@ -2491,6 +2922,251 @@
...
             ]
+        },
+        "NorthSouthHardeningsPolicySummary": {
+            "description": "North-south hardening policy summary",
+            "properties": {
+                "id": {
+                    "description": "The north-south policy id",
+                    "type": "string"
+                },
+                "active": {
```
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: please make it an enum. Its more descriptive and allows for future addition of states.

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json [hunk 5]
```diff
@@ -2491,6 +2922,251 @@
...
+                    "items": {
+                        "type": "string"
+                    }
+                }
+            }
+        },
+        "NorthSouthHardeningsPolicyProperties": {
+            "description": "A north-south hardening policy",
+            "properties": {
+                "active": {
```
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: same as above about string enum instead of bool

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json [hunk 6]
```diff
@@ -2491,6 +2922,251 @@
...
+                "nextLink": {
+                    "description": "The URL to get the next set of results",
+                    "type": "string"
+                }
+            }
+        },
+        "NorthSouthHardeningsPolicyRule": {
+            "description": "A north-south hardening policy rule",
+            "properties": {
+                "name": {
```
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: name is already there as a top level property, why is this needed?

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/examples/NorthSouthHardenings/ListNorthSouthHardenings_example.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                "value": [
+                    {
+                        "id": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourcegroups/rg1/providers/Microsoft.Security/northSouthHardenings/vm1",
+                        "name": "vm1",
+                        "type": "Microsoft.Security/northSouthHardenings",
+                        "properties": {
+                            "protectedResourceId": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourcegroups/rg1/providers/Microsoft.Compute/virtualMachines/vm1",
+                            "policies": [
+                                {
+                                    "id": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourcegroups/rg1/providers/Microsoft.Security/northSouthHardening/vm1/policies/recommended",
```
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/examples/NorthSouthHardenings/ListNorthSouthHardenings_example.json]: what are the possible names for policy? can the user give any name when creating? use of "default" in the name makes me want to double check.

Comment (reamkazaz) [reply specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/examples/NorthSouthHardenings/ListNorthSouthHardenings_example.json]: We create a default policy in the background and its name will be "recommended".
A user can create or update other policies but not the "recommended" one

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json (hunk 5 above)
Comment (reamkazaz) [reply specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: done

File: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json (hunk 4 above)
Comment (reamkazaz) [reply specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json]: done

**Comment by ravbhatnagar** [comment]
looks good. signing off from ARM side.

**Comment by anuchandy** [comment]
@reamkazaz this PR is idle state for last 2+ months, did we deploy the service? if not I would suggest closing this PR and re-open it at later point.
